src/mx25r6435f_driver.c

```diff
@@ -0,0 +1,1174 @@
+    the flag status register program erase controller bit is set to 0. This command is ignored
+    if the device is not in a suspended state.
+    */
+
+    if (BSP_QSPI_GetStatus() == QSPI_BUSY)
+    {
+      return QSPI_OK;
+    }
+
+    return QSPI_ERROR;
+  }
+
+  return QSPI_OK;
+}
+
+/**
+  * @brief  This function enter the QSPI memory in deep power down mode.
+  * @retval QSPI memory status
+  */
+uint8_t BSP_QSPI_EnterDeepPowerDown(void)
+{
+  QSPI_CommandTypeDef sCommand;
+
+  /* Initialize the deep power down command */
+  sCommand.InstructionMode   = QSPI_INSTRUCTION_1_LINE;
+  sCommand.Instruction       = DEEP_POWER_DOWN_CMD;
+  sCommand.AddressMode       = QSPI_ADDRESS_NONE;
+  sCommand.AlternateByteMode = QSPI_ALTERNATE_BYTES_NONE;
+  sCommand.DataMode          = QSPI_DATA_NONE;
+  sCommand.DummyCycles       = 0;
+  sCommand.DdrMode           = QSPI_DDR_MODE_DISABLE;
+  sCommand.DdrHoldHalfCycle  = QSPI_DDR_HHC_ANALOG_DELAY;
+  sCommand.SIOOMode          = QSPI_SIOO_INST_EVERY_CMD;
+
+  /* Send the command */
+  if (HAL_QSPI_Command(&QSPIHandle, &sCommand, HAL_QPSI_TIMEOUT_DEFAULT_VALUE) != HAL_OK)
+  {
+    return QSPI_ERROR;
+  }
+
+  /* ---          Memory takes 10us max to enter deep power down          --- */
+  /* --- At least 30us should be respected before leaving deep power down --- */
+
+  return QSPI_OK;
+}
+
+/**
+  * @brief  This function leave the QSPI memory from deep power down mode.
+  * @retval QSPI memory status
+  */
+uint8_t BSP_QSPI_LeaveDeepPowerDown(void)
+{
+  QSPI_CommandTypeDef sCommand;
+
+  /* Initialize the erase command */
+  sCommand.InstructionMode   = QSPI_INSTRUCTION_1_LINE;
+  sCommand.Instruction       = NO_OPERATION_CMD;
+  sCommand.AddressMode       = QSPI_ADDRESS_NONE;
+  sCommand.AlternateByteMode = QSPI_ALTERNATE_BYTES_NONE;
+  sCommand.DataMode          = QSPI_DATA_NONE;
+  sCommand.DummyCycles       = 0;
+  sCommand.DdrMode           = QSPI_DDR_MODE_DISABLE;
+  sCommand.DdrHoldHalfCycle  = QSPI_DDR_HHC_ANALOG_DELAY;
+  sCommand.SIOOMode          = QSPI_SIOO_INST_EVERY_CMD;
+
+  /* Send the command */
+  if (HAL_QSPI_Command(&QSPIHandle, &sCommand, HAL_QPSI_TIMEOUT_DEFAULT_VALUE) != HAL_OK)
+  {
+    return QSPI_ERROR;
+  }
+
+  /* --- A NOP command is sent to the memory, as the nCS should be low for at least 20 ns --- */
+  /* ---                  Memory takes 35us min to leave deep power down                  --- */
+
+  return QSPI_OK;
+}
+
+/**
+  * @brief  Initializes the QSPI MSP.
+  * @retval None
+  */
+__weak void BSP_QSPI_MspInit(void)
+{
+  GPIO_InitTypeDef GPIO_InitStruct;
+  GPIO_TypeDef *port;
+  uint8_t idx;
+  PinName pin;
+
+  /* Enable the QuadSPI memory interface clock */
+  __HAL_RCC_QSPI_CLK_ENABLE();
+
+  /* Reset the QuadSPI memory interface */
+  __HAL_RCC_QSPI_FORCE_RESET();
+  __HAL_RCC_QSPI_RELEASE_RESET();
+
+  /* QSPI CLK, CS, D0, D1, D2 and D3 GPIO pins configuration  */
+  for(idx = 0; idx < 6; idx++)

```

Comment:
Maybe iterate until PinMap_QUADSPI[idx].pin == NC ? and so remove 
+    if(pin == NC)
+      break;

Reply:
I didn't do that in case the user forgets to comment a line in PinMap_QUADSPI[]. So I don't know how the QSPI peripheral will react if there is more than 6 pins configured.
I don't know what do you think? Maybe do we leave the responsibility to the user?

Reply:
Even if he forgets to comment lines, this will configure unrequired pins for QUADSPI but at least the good one will be initialized and those not used by the QUADSPI will break his use case if pins are used for other feature.
But sure it's up to the user to ensure hardware configs is correct.